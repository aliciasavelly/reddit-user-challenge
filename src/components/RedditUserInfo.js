import React from 'react';

import axios from 'axios';
import InfoIndex from './InfoIndex';

export default class RedditUserInfo extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			username: '',
			userComments: [],
			userPosts: [],
			shownInfo: 'posts',
			error: '',
		};
		this.updateUsername = this.updateUsername.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	updateUsername(e) {
		this.setState({ username: e.target.value });
	}

	handleSubmit(e) {
		e.preventDefault();
		const { username } = this.state;
		axios.get(`https://www.reddit.com/user/${username}/submitted.json`)
			.then((postsResponse) => {
				const posts = [];
				postsResponse.data.data.children.forEach((postData) => {
					const userPost = {};
					userPost.title = postData.data.title;
					userPost.score = postData.data.score;
					userPost.url = postData.data.url;
					posts.push(userPost);
				});
				this.setState({ userPosts: posts, error: '' });
			})
			.catch((error) => {
				console.log(error);
				this.setState({ 
					error: 'Unable to fetch the data; please input a valid username',
				});
			});
			
		axios.get(`https://www.reddit.com/user/${username}/comments.json`)
			.then((commentsResponse) => {
				const comments = [];
				commentsResponse.data.data.children.forEach((commentData) => {
					const userComment = {};
					userComment.body = commentData.data.body;
					userComment.score = commentData.data.score;
					userComment.url = commentData.data.link_url;
					comments.push(userComment);
				});
				this.setState({ userComments: comments, error: '' });
			})
			.catch((error) => {
				console.log(error);
				this.setState({ 
					error: 'Unable to fetch the data; please input a valid username', 
				});
			});
	}
	

	shownInfo() {
		const { userPosts, userComments, username, shownInfo } = this.state;

		let index;
		if (shownInfo === "posts") {
			index = userPosts;
		} else {
			index = userComments;
		}

		const props = {
			index,
			shownInfo,
			username,
		};

		return <InfoIndex {...props} />;
	}

	toggle(option) {
		return () => {
			this.setState({ shownInfo: option });
		};
	}

	render() {
		let postsTabClass = "tab";
		let commentsTabClass = "tab";

		if (this.state.shownInfo === "posts") {
			postsTabClass = "tab selected";
		} else {
			commentsTabClass = "tab selected";
		}
		return(
			<div className="reddit-info">
				<form className="username-form" onSubmit={this.handleSubmit}>
					<label>
						Enter Username:
						<input 
							type="text" 
							name="username" 
							onChange={this.updateUsername}
						/>
					</label>
					<input type="submit" value="Submit" />
					<div className="error">
						{ this.state.error }
					</div>
				</form>
				<div className="tabs">
					<div className={postsTabClass} onClick={this.toggle('posts')}>
						Posts
					</div>
					<div className={commentsTabClass} onClick={this.toggle('comments')}>
						Comments
					</div>
				</div>
				{ this.shownInfo() }
			</div>
		);
	}
}
           