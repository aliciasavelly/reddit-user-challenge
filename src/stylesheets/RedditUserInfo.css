.username-form {
  text-align: center;
  padding: 25px 25px 40px 25px;
  background: #C3DBF7;
}

label input {
  margin: 0 10px;
  padding: 7px;
  border: none;
}

.error {
	margin-top: 10px;
	color: red;
}

.tabs {
	padding-left: 10px;
	background: #C3DBF7;
	display: flex;
}	

.tab {
	margin-left: 5px;
	background: #EBF5FF;
	padding: 5px;
	font-weight: bold;
}

.selected {
	background: white;
	color: red;
	border-bottom: none;
}