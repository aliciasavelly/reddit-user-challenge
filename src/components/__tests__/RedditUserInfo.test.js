import React from 'react';
import ReactDOM from 'react-dom';
import RedditUserInfo from '../RedditUserInfo';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('RedditUserInfo', () => {
	const RedditUserInfoWrapper = shallow(<RedditUserInfo />);
	const usernameInput = RedditUserInfoWrapper.find('input[type="text"]');

	it('updates the username on input', () => {
		usernameInput.simulate('change', { target: { value: 'a' }});
		expect(RedditUserInfoWrapper.state().username).toEqual('a');
	});

	it('renders the username input and tabs', () => {
		expect(shallowToJson(RedditUserInfoWrapper)).toMatchSnapshot();
	});
});



