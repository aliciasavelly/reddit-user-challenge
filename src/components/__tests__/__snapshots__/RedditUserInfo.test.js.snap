// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedditUserInfo renders the username input and tabs 1`] = `
<div
  className="reddit-info"
>
  <form
    className="username-form"
    onSubmit={[Function]}
  >
    <label>
      Enter Username:
      <input
        name="username"
        onChange={[Function]}
        type="text"
      />
    </label>
    <input
      type="submit"
      value="Submit"
    />
    <div
      className="error"
    />
  </form>
  <div
    className="tabs"
  >
    <div
      className="tab selected"
      onClick={[Function]}
    >
      Posts
    </div>
    <div
      className="tab"
      onClick={[Function]}
    >
      Comments
    </div>
  </div>
  <Component
    index={Array []}
    shownInfo="posts"
    username="a"
  />
</div>
`;
